src/sorting/src/tests/quick_sort_test.cairo

```diff
@@ -0,0 +1,175 @@
+use alexandria_data_structures::vec::{Felt252Vec, VecTrait};
+use alexandria_sorting::{is_equal, quick_sort};
+use core::option::OptionTrait;
+
+
+#[test]
+#[available_gas(20000000000000)]
+fn quicksort_test() {
+    let mut data = VecTrait::<Felt252Vec, u32>::new();
+    data.push(4_u32);
+    data.push(2_u32);
+    data.push(1_u32);
+    data.push(3_u32);
+    data.push(5_u32);
+    data.push(0_u32);
+    let mut correct = array![0_u32, 1_u32, 2_u32, 3_u32, 4_u32, 5_u32];
+
+    let mut sorted = quick_sort::quick_sort(data);
+
+    let mut compare_id = 0;
+    let array_len = correct.len();
+    loop {
+        if compare_id == array_len {
+            break;
+        }
+
+        assert(sorted[compare_id] == *correct[compare_id], 'invalid result');
+
+        compare_id += 1;
+    }
```

Comment:
move that into a helper function it'll be easier to refact

Reply:
Fixed ;)